docs/conceptual/runtime/contracts/ink.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ink!
+---
+
+ink! is a [Rust](https://www.rust-lang.org/)-based embedded domain specific language
+([eDSL](https://wiki.haskell.org/Embedded_domain_specific_language)) for writing
+[Wasm](https://webassembly.org/) smart contracts specifically for the [SRML Contracts module](conceptual/runtime/contracts/contracts_module.md).
+
+## Design
+
+The main goals of ink! are user friendliness, conciseness, and efficiency.
+
+ink! is designed to be as close to the Rust programming language as possible. The language uses
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag standard Rust structures into understandable contract components.
+
+```rust
+#[ink(...)]
+```
+
+Because ink! follows Rust standards, tools like `rustfmt` and `rust-analyzer` already work out of
+the box.
+
+### Contract Components
+
+ink! should still feel familiar to developers who have programmed using other modern smart contract
+languages. The skeleton of a contract has all of the same components that you might expect:
+
+  * Events - `#[ink(event)]`
+    * Event Topics - `#[ink(topic)]`
+  * Storage - `#[ink(storage)]`
+  * Constructor Functions - `#[ink(constructor)]`
+  * Message Functions - `#[ink(message)]`
```

Comment:
Not sure if it is useful to add those `#[ink(..)]` at this point as it might confuse readers.

Reply:
Removed them